pkg/cmd/util/clientcmd/factory.go

```diff
@@ -569,6 +539,59 @@ func (w *Factory) ApproximatePodTemplateForObject(object runtime.Object) (*api.P
 	}
 }
 
+// PodForRemoteShell resolves the passed resource and returns
+func (f *Factory) PodForRemoteShell(resource string) (string, error) {
```

Comment:
this is really just "pod for object"... it doesn't have to be for a remote shell